daemon/oci_linux.go

```diff
@@ -813,6 +814,30 @@ func WithDevices(daemon *Daemon, c *container.Container) coci.SpecOpts {
 			for _, d := range hostDevices {
 				devs = append(devs, oci.Device(d))
 			}
+
+			// adding device mappings in privileged containers
+			for _, deviceMapping := range c.HostConfig.Devices {
+				// issue a warning that custom cgroup permissions are ignored in privileged mode
+				if deviceMapping.CgroupPermissions != "rwm" {
+					logrus.WithField("container", c.ID).Warnf("custom %s permissions for device %s are ignored in privileged mode", deviceMapping.CgroupPermissions, deviceMapping.PathOnHost)
+				}
+				// issue a warning that the device path already exists via /dev mounting in privileged mode
+				if deviceMapping.PathOnHost == deviceMapping.PathInContainer {
+					logrus.WithField("container", c.ID).Warnf("path in container %s already exists in privileged mode", deviceMapping.PathInContainer)
+					continue
+				}
+				// check if the path exists in the container. need to create a device only if the
+				// path does not exists.
+				if _, err := os.Stat(deviceMapping.PathInContainer); !os.IsNotExist(err) {
```

Comment:
@kolyshkin Just now I was testing this change. This code where we do an `os.Stat` for a path in the container. Is it the right way to do it. Because I can see that the container creation fails, if such a path exists on the host.

Example: There is a `/test` directory on the host. And the mapping used is `--device /dev/sdc:/test`. This will fail because `os.Stat` checks and finds that a directory exists, but in actual case there is no such directory inside the container.

The same issue can happen in reverse also. There is a directory `/test` available inside the container but not on the host. When `docker run` is executed with `--device /dev/sdc:/test`, it won't show any errors. Inside the container, the device won't be available at `/test`. Just the original directory. 

How should both these cases be handled.
cc: @thaJeztah